### PR TITLE
[Intel MKL] Supporting native format in Convolution Fwd

### PR DESCRIPTION
**Pull request by mahmoud-abuzaina**
This PR adds support for native(user) format in convolution forward op. _MklNative* ops will be used in both graph and eager modes. 
This PR needs to be merged after "DNNL 1.5.1 Upgrade" PR is merged #42073

### CODE REVIEW

File: tensorflow/core/kernels/mkl/mkl_conv_ops.cc [hunk 1]
```diff
@@ -77,20 +78,23 @@ struct MklConvFwdParams {
...
-        padding_right(padding_right) {}
+        padding_right(padding_right),
+        tf_fmt(tf_fmt) {}
 };
 
 // With quantization, input, filter, and output can have different types
 // so we use different template parameter for each type
-template <typename Tinput, typename Tfilter, typename Tbias, typename Toutput>
+template <typename Tinput, typename Tfilter, typename Tbias, typename Toutput,
+          bool native_format>
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl/mkl_conv_ops.cc:97]: Same comment as in #42072, I don't think `native_format` needs to be a template parameter here. Can we make it a normal parameter?

File: tensorflow/core/kernels/mkl/mkl_conv_ops.cc [hunk 2]
```diff
@@ -360,29 +370,31 @@ class MklConvFwdPrimitive : public MklPrimitive {
 // TODO(nhasabni): We should not require passing a type to MklPrimitiveFactory.
 // But removing the need for type in MklPrimitiveFactory is going to require
 // change to every MKL op. So not doing it now. Instead passing float.
-template <typename Tinput, typename Tfilter, typename Tbias, typename Toutput>
+template <typename Tinput, typename Tfilter, typename Tbias, typename Toutput,
+          bool native_format>
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl/mkl_conv_ops.cc:374]: Same. Can we make `native_format` a normal parameter?

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR and I'm sorry for the delay!

File: tensorflow/core/kernels/mkl/mkl_conv_ops.cc (hunk 1 above)
Comment (mahmoud-abuzaina) [reply tensorflow/core/kernels/mkl/mkl_conv_ops.cc]: Done.

File: tensorflow/core/kernels/mkl/mkl_conv_ops.cc (hunk 2 above)
Comment (mahmoud-abuzaina) [reply tensorflow/core/kernels/mkl/mkl_conv_ops.cc]: Done.

**Comment by mahmoud-abuzaina** [comment]
Thank you for your review. I have addressed the comments.

**Review by penpornk — APPROVED** [review]
Thank you for the quick changes!

**Pull request merged**